### PR TITLE
[JS] Extend documentation sevice binding transformation

### PR DESCRIPTION
**Pull request by FrankEssenberger**
## What Has Changed?

see title

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)

### CODE REVIEW

File: docs/js/features/connectivity/destination.mdx [hunk 1]
```diff
@@ -190,27 +190,29 @@ We have a separate [guide](./destination-cache-isolation.mdx) on the cache optio
...
 
-The service credentials, in other words `VCAP_SERVICES` environment variables, may contain a destination.
-If you want to use this information, the name of the service instance must be provided as the destination name.
-Currently, two services types are supported out of the box `business-logging` and `s4-hana-cloud` with the following transformation functions:
+The service credentials, in other words `VCAP_SERVICES` environment variables represent services bound to the destination.
+If you want to consume such a service the SAP Cloud SDK can create a destination from the service binding for you.
+You do not need to create a dedicated destination in the destination service.
+Provide the name of the service instance as the destination name and the SAP Cloud SDK will:
+- Find the service binding with the given name
+- Extract the URL form the service binding
```
Comment (florian-richter) [inline docs/js/features/connectivity/destination.mdx:198]:
```suggestion
- Extract the URL from the service binding
```
[req] typo

File: docs/js/features/connectivity/destination.mdx [hunk 2]
```diff
@@ -190,27 +190,29 @@ We have a separate [guide](./destination-cache-isolation.mdx) on the cache optio
...
+Currently, 5 services types are supported out of the box:
+
+- `business-logging`
+- `s4-hana-cloud`
+- `saas-registry`
+- `workflow`
+- `destination`
+
+If you need a destination for a different service create a custom `serviceBindingTransformFn` and pass it with `destinationFetchOptions` or consider creating a [feature request](https://github.com/SAP/cloud-sdk-js/issues/new?assignees=&labels=feature+request&template=feature_request.md&title=).
+Contributions providing transform function are highly welcome.
```
Comment (florian-richter) [inline docs/js/features/connectivity/destination.mdx:211]:
```suggestion
Contributions providing transform functions are highly welcome.
```
[req] possibly a typo

**Review by florian-richter — CHANGES_REQUESTED** [review]
Some minor fixes necessary

**Review by deekshas8 — COMMENTED** [review]
Minor comments

**Review by florian-richter — APPROVED** [review]

**Pull request merged**